src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -503,17 +503,17 @@ class ResourceService(
       samRequestContext: SamRequestContext
   ): IO[List[Boolean]] =
     accessPolicyDAO.listAccessPolicies(resourceId, samRequestContext) flatMap { policiesForResource =>
```

Comment:
some minor logic changes below. technically, every managed group has a public policy (the admin notifiers policy). so with the previous implementation, it was impossible to leave a managed group because it looked for the presence of a single public policy and rejected the request if it found one. the new logic checks to see if the user has additional policies on the resource beyond the public policy, and allows the user to leave them if so.